### DISCUSSION TITLE
`init_alias` equivalent in nanobind?

### DISCUSSION THREAD

**Discussion by zhqrbitee** [wjakob/nanobind#176] [Q&A] [answered]
First, thanks for the great work on nanobind! I'm currently evaluating how to migrate from pybind11 to nanobind for my project and it seems promising. 
However, in pybind11, we used `init_alias` to tell pybind11 to always initialize the trampoline class when creating a new instance: https://pybind11.readthedocs.io/en/stable/advanced/classes.html#extended-trampoline-class-functionality. After searching through nanobind's docs, I didn't find any equivalent of `init_alias`. So is it a missing feature in nanobind now, or simply I miss it in the docs?

**Comment by wjakob** [comment, marked as answer]
Nanobind always creates the alias class at the moment (it is as if `init_alias` was specified for all constructors). We may consider porting this optimization if it is found to be a performance issue.

**Reply by zhqrbitee** [reply]
Thanks for the answer!

**Comment by zhqrbitee** [comment]
@wjakob A follow up question, is there any plan to support the optimization from pybind11 (that don't create alias class if python class doesn't actually inherit)?

**Reply by wjakob** [reply]
> @wjakob A follow up question, is there any plan to support the optimization from pybind11 (that don't create alias class if python class doesn't actually inherit)?

I added it in the meantime. The feature to optionally disable the optimization was not ported (yet).
